lib/spack/spack/test/stage.py

```diff
@@ -902,8 +917,91 @@ def test_diystage_preserve_file(self, tmpdir):
         with open(readmefn) as _file:
             _file.read() == _readme_contents
 
+    def test_create_cmakebuildstage(self, cmake_ext_build, mock_stage_archive):
+        """Test that cmake external build dir is created during stage creation"""
+        archive = mock_stage_archive()
+        with CMakeBuildStage("jk6dj", archive.url, name=self.stage_name) as stage:
+            # external build dir should already have been created at this point
+            # ensure symlink to build dir exists
+            assert str(stage.root_stage_context.name) == "spack-build-jk6dj"
+            assert "spack-build-jk6dj" in os.listdir(stage.path)
+            # check symlink points to the right place
+            assert str(stage.root_stage_context.resolve()) == str(cmake_ext_build / "a")
+            # ensure ext build dir was set honoring config
+            assert spack.config.get("config:cmake_ext_build") == str(stage._remote_stage.parent)
```

Comment:
A couple minor requests:

* "Canonicalize" the configuration value by constructing a `Path` with it (avoid fragility due to string comparisons with paths with/without trailing `/`)
* swap order here (in general stick with `assert the-value-I-calculated == the-thing-I-expect-it-to-be`)